### PR TITLE
common/model/transaction - logical fields separation

### PR DESCRIPTION
**Pull request by tsvisabo**
- logical fields separation
- remove hard-coded TRANSACTION/TEST_TRANSACTION/TRITS
- will follow storage loading methods that make use on some fields only

### CODE REVIEW

**Review by th0br0 — APPROVED** [review]

File: common/model/transaction.h [hunk 1]
```diff
@@ -85,106 +92,262 @@ struct _iota_transaction {
...
+  // 27 trytes = 81 trits
+  flex_trit_t tag[FLEX_TRIT_SIZE_81];
+};
+
+struct _iota_transaction_fields_consensus {
   // 81 trytes = 243 trits
   flex_trit_t hash[FLEX_TRIT_SIZE_243];
-  // Total 2754 trytes
+  // 81 trytes = 243 trits
+  flex_trit_t bundle[FLEX_TRIT_SIZE_243];
```
Comment (th0br0) [inline common/model/transaction.h]: let's move the bundle back to the essence for now?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: common/model/transaction.h [hunk 2]
```diff
@@ -54,9 +54,17 @@ extern "C" {
  * Transaction data structure
  ***********************************************************************************************************/
 typedef struct _iota_transaction *iota_transaction_t;
-struct _iota_transaction {
-  // 2187 trytes = 6561 trits
-  flex_trit_t signature_or_message[FLEX_TRIT_SIZE_6561];
+typedef struct _iota_transaction_fields_essence
+    iota_transaction_fields_essence_t;
+typedef struct _iota_transaction_fields_attachement
```
Comment (thibault-martinez) [inline common/model/transaction.h]: Typo

File: common/model/transaction.h [hunk 3]
```diff
@@ -54,9 +54,17 @@ extern "C" {
...
+    iota_transaction_fields_attachement_t;
```
Comment (thibault-martinez) [inline common/model/transaction.h]: Typo

File: common/model/transaction.h (hunk 1 above)
Comment (tsvisabo) [reply common/model/transaction.h]: done

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by th0br0 — APPROVED** [review]

**Pull request merged**